lib/whitehall_importer/create_edition.rb

```diff
@@ -96,6 +98,28 @@ def set_withdrawn_status(edition)
       edition.update!(status: build_status("withdrawn", withdrawal))
     end
 
+    def create_scheduled_edition
+      create_edition(status: build_status(pre_scheduled_status),
+                     current: current,
+                     edition_number: edition_number).tap { |edition| set_scheduled_status(edition) }
+    end
+
+    def pre_scheduled_status
+      history.submitted_before_scheduling? ? "submitted_for_review" : "draft"
+    end
+
+    def set_scheduled_status(edition)
+      raise AbortImportError, "Cannot create scheduled status without scheduled_publication" if whitehall_edition["scheduled_publication"].blank?
+
+      scheduling = Scheduling.new(
+        pre_scheduled_status: edition.status,
+        reviewed: !whitehall_edition["force_published"],
+        publish_time: whitehall_edition["scheduled_publication"],
+      )
+
+      edition.update!(status: build_status("scheduled", scheduling))
+    end
```

Comment:
I wonder if these 3 methods will read nicer as a single one, since not that much actually goes on in them and that would save us from a variety of scheduled methods methods only used in this context.

```
def create_scheduled_edition
  raise AbortImportError, "Cannot create scheduled status without scheduled_publication" if whitehall_edition["scheduled_publication"].blank?

  pre_scheduled_state = history.state_event("submitted") ? "submitted_for_review" : "draft"
  edition = create_edition(status: build_status(pre_scheduled_state), 
                           current: current, 
                           edition_number: edition_number)

  scheduling = Scheduling.new(pre_scheduled_status: status,
                              reviewed: !whitehall_edition["force_published"],
                              publish_time: whitehall_edition["scheduled_publication"])
  edition.update!(status: build_status("scheduled", scheduling)
end
```

What do you think?